src/material-experimental/mdc-list/testing/selection-list-harness.ts

```diff
@@ -0,0 +1,145 @@
+  static hostSelector = '.mat-mdc-selection-list';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatSelectionListHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which selection list instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: SelectionListHarnessFilters = {}):
+      HarnessPredicate<MatSelectionListHarness> {
+    return new HarnessPredicate(MatSelectionListHarness, options);
+  }
+
+  _itemHarness = MatListOptionHarness;
+
+  /** Whether the selection list is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return await (await this.host()).getAttribute('aria-disabled') === 'true';
+  }
+
+  /**
+   * Selects all items matching any of the given filters.
+   * @param filters Filters that specify which items should be selected.
+   */
+  async selectItems(...filters: ListOptionHarnessFilters[]): Promise<void> {
+    const items = await this._getItems(filters);
+    await Promise.all(items.map(item => item.select()));
+  }
+
+  /**
+   * Deselects all items matching any of the given filters.
+   * @param filters Filters that specify which items should be deselected.
+   */
+  async deselectItems(...filters: ListItemHarnessFilters[]): Promise<void> {
+    const items = await this._getItems(filters);
+    await Promise.all(items.map(item => item.deselect()));
+  }
+
+  /** Gets all items matching the given list of filters. */
+  private async _getItems(filters: ListOptionHarnessFilters[]): Promise<MatListOptionHarness[]> {
+    if (!filters.length) {
+      return this.getItems();
+    }
+    return ([] as MatListOptionHarness[])
+        .concat(...await Promise.all(
+            filters.map(filter => this.locatorForAll(MatListOptionHarness.with(filter))())));
+  }
+}
+
+/** Harness for interacting with a MDC-based list option. */
+export class MatListOptionHarness extends MatListItemHarnessBase {
+  /** The selector for the host element of a `MatListOption` instance. */
+  static hostSelector = '.mat-mdc-list-option';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatListOptionHarness` that
+   * meets certain criteria.
+   * @param options Options for filtering which list option instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: ListOptionHarnessFilters = {}): HarnessPredicate<MatListOptionHarness> {
+    return getListItemPredicate(MatListOptionHarness, options)
+        .addOption(
+            'is selected', options.selected,
+            async (harness, selected) => await harness.isSelected() === selected);
+  }
+
+  private _beforeCheckbox = this.locatorForOptional('.mdc-list-item__graphic .mdc-checkbox');
+
+  /** Gets the position of the checkbox relative to the list option content. */
+  async getCheckboxPosition(): Promise<'before'|'after'> {
```

Comment:
Should we have a type for `'before' | 'after'`? It seems to be repeated in a lot of places and it'll be a pain if we decide to add another one.

Reply:
I agree. Created a type for both implementations of the selection list. The non-MDC list harness has the same code.